kroxylicious/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/ClusterNetworkAddressConfigProviderContributorManager.java

```diff
@@ -5,50 +5,40 @@
  */
 package io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider;
 
-import java.util.Iterator;
-import java.util.ServiceLoader;
-
 import io.kroxylicious.proxy.clusternetworkaddressconfigprovider.ClusterNetworkAddressConfigProviderContributor;
 import io.kroxylicious.proxy.config.BaseConfig;
 import io.kroxylicious.proxy.service.ClusterNetworkAddressConfigProvider;
+import io.kroxylicious.proxy.service.ContributionManager;
 
 import static io.kroxylicious.proxy.service.Context.wrap;
 
+@SuppressWarnings("java:S6548")
 public class ClusterNetworkAddressConfigProviderContributorManager {
 
-    private static final ClusterNetworkAddressConfigProviderContributorManager INSTANCE = new ClusterNetworkAddressConfigProviderContributorManager();
-
-    private final ServiceLoader<ClusterNetworkAddressConfigProviderContributor> contributors;
+    public static final ClusterNetworkAddressConfigProviderContributorManager INSTANCE = new ClusterNetworkAddressConfigProviderContributorManager();
 
     private ClusterNetworkAddressConfigProviderContributorManager() {
-        this.contributors = ServiceLoader.load(ClusterNetworkAddressConfigProviderContributor.class);
     }
 
     public static ClusterNetworkAddressConfigProviderContributorManager getInstance() {
         return INSTANCE;
     }
 
     public Class<? extends BaseConfig> getConfigType(String shortName) {
-        Iterator<ClusterNetworkAddressConfigProviderContributor> it = contributors.iterator();
-        while (it.hasNext()) {
-            ClusterNetworkAddressConfigProviderContributor contributor = it.next();
-            Class<? extends BaseConfig> configType = contributor.getConfigType(shortName);
-            if (configType != null) {
-                return configType;
-            }
+        try {
+            return ContributionManager.INSTANCE.getDefinition(ClusterNetworkAddressConfigProviderContributor.class, shortName).configurationType();
+        }
+        catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("No endpoint provider found for name '" + shortName + "'");
         }
-
-        throw new IllegalArgumentException("No endpoint provider found for name '" + shortName + "'");
     }
 
     public ClusterNetworkAddressConfigProvider getClusterEndpointConfigProvider(String shortName, BaseConfig baseConfig) {
-        for (ClusterNetworkAddressConfigProviderContributor contributor : contributors) {
-            var assigner = contributor.getInstance(shortName, wrap(baseConfig));
-            if (assigner != null) {
-                return assigner;
-            }
+        try {
+            return ContributionManager.INSTANCE.getInstance(ClusterNetworkAddressConfigProviderContributor.class, shortName, wrap(baseConfig));
+        }
+        catch (IllegalArgumentException e) {
```

Comment:
I'm not convinced by catch and re-throw, but it provides much better error messages.